pkg/apis/metrics/types.go

```diff
@@ -16,14 +16,427 @@ limitations under the License.
+}
+
+type TcpStat struct {
+	// Number of connections in the Established state.
+	Established uint64 `json:"established,omitempty"`
+	// Number of connections in the SynSent state.
+	SynSent uint64 `json:"synSent,omitempty"`
+	// Number of connections in the SynRecv state.
+	SynRecv uint64 `json:"synRecv,omitempty"`
+	// Number of connections in the FinWait1 state.
+	FinWait1 uint64 `json:"finWait1,omitempty"`
+	// Number of connections in the FinWait2 state.
+	FinWait2 uint64 `json:"finWait2,omitempty"`
+	// Number of connections in the TimeWait state.
+	TimeWait uint64 `json:"timeWait,omitempty"`
+	// Number of connections in the Close state.
+	Close uint64 `json:"close,omitempty"`
+	// Number of connections in the CloseWait state.
+	CloseWait uint64 `json:"closeWait,omitempty"`
+	// Number of connections in the LastAck state.
+	LastAck uint64 `json:"lastAck,omitempty"`
+	// Number of connections in the Listen state.
+	Listen uint64 `json:"listen,omitempty"`
+	// Number of connections in the Closing state.
+	Closing uint64 `json:"closing,omitempty"`
+}
+
+type NetworkStats struct {
+	// Network stats by interface.
+	Interfaces []InterfaceStats `json:"interfaces,omitempty"`
+	// TCP connection stats (Established, Listen...)
+	Tcp TcpStat `json:"tcp,omitempty"`
+	// TCP6 connection stats (Established, Listen...)
+	Tcp6 TcpStat `json:"tcp6,omitempty"`
+}
+
+type InterfaceStats struct {
+	// The name of the interface.
+	Name string `json:"name,omitempty"`
+	// Cumulative count of bytes received.
+	RxBytes uint64 `json:"rxBytes,omitempty"`
+	// Cumulative count of packets received.
+	RxPackets uint64 `json:"rxPackets,omitempty"`
+	// Cumulative count of receive errors encountered.
+	RxErrors uint64 `json:"rxErrors,omitempty"`
+	// Cumulative count of packets dropped while receiving.
+	RxDropped uint64 `json:"rxDropped,omitempty"`
+	// Cumulative count of bytes transmitted.
+	TxBytes uint64 `json:"txBytes,omitempty"`
+	// Cumulative count of packets transmitted.
+	TxPackets uint64 `json:"txPackets,omitempty"`
+	// Cumulative count of transmit errors encountered.
+	TxErrors uint64 `json:"txErrors,omitempty"`
+	// Cumulative count of packets dropped while transmitting.
+	TxDropped uint64 `json:"txDropped,omitempty"`
+}
+
+// Instantaneous CPU stats
+type CpuInstStats struct {
+	Usage CpuInstUsage `json:"usage,omitempty"`
+}
+
+// CPU usage time statistics.
+type CpuInstUsage struct {
+	// Total CPU usage.
+	// Units: nanocores per second
+	Total uint64 `json:"total,omitempty"`
+
+	// Per CPU/core usage of the container.
+	// Unit: nanocores per second
+	PerCpu []uint64 `json:"perCpu,omitempty"`
+
+	// Time spent in user space.
+	// Unit: nanocores per second
+	User uint64 `json:"user,omitempty"`
+
+	// Time spent in kernel space.
+	// Unit: nanocores per second
+	System uint64 `json:"system,omitempty"`
+}
+
+// CPU usage time statistics.
+type CpuUsage struct {
+	// Total CPU usage.
+	// Units: nanoseconds
```

Comment:
But if it uses 2 cpus for 5 seconds, isn't it `10*10^9`? So shouldn't it be core-nanoseconds == nanocore-seconds?